app/Http/Controllers/SearchController.php

```diff
@@ -37,7 +39,11 @@ public function advanced(Request $request)
             $hymnTranslations = collect( [] );
         }
 
-        $people = Person::orderBy('display_name')->get();
+        $people = Person::leftJoin('hymns', 'received_by', '=', 'persons.id')
```

Comment:
I like this idea and I wonder about a) restricting the list to people who have at least one, and b) doing separate lists for the received_by list and the offered_to list.